File: src/main/java/com/example/dpop/orchestrator/api/v1/AttemptRequest.java
package com.example.dpop.orchestrator.api.v1;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record AttemptRequest(
        String method,
        Map<String, Object> data
) {}

File: src/main/kotlin/com/example/dpop/orchestrator/api/v1/AttemptRequest.kt
package com.example.dpop.orchestrator.api.v1

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
@JvmRecord
data class AttemptRequest(
    val method: String?,
    val mode: String?,
    val data: Map<String, Any>?
) {
    constructor(method: String, data: Map<String, Any>?) : this(method, null, data)
}
